src/gamma_inc.jl

```diff
@@ -0,0 +1,580 @@
+    elseif x <= 0.4
+        u = (x-0.375)/0.375
+        return _log1pmx_ker(u) - 3.55829253011726237e-1 + 0.625*u
+    elseif x <= 0.6
+        u = 2.0*(x-0.5)
+        return _log1pmx_ker(u) - 1.93147180559945309e-1 + 0.5*u
+    else
+        return log1pmx(x - 1.0)
+    end
+end
+
+"""
+   rgamma1pm1(a)
+
+   Computation of 1/Gamma(a+1) - 1 for -0.5<=a<=1.5 : ``1/\\Gamma (a+1) - 1``
+   Uses the relation gamma(a+1) = a*gamma(a)
+"""
+function rgamma1pm1(a::Float64)
+    t=a
+    rangereduce = a > 0.5
+    t = rangereduce ? a-1 : a #-0.5<= t <= 0.5
+    if t == 0.0
+        return 0.0
+    elseif t < 0.0        
+        top = @horner(t , -.422784335098468E+00 , -.771330383816272E+00 , -.244757765222226E+00 , .118378989872749E+00 , .930357293360349E-03 , -.118290993445146E-01 , .223047661158249E-02 , .266505979058923E-03 , -.132674909766242E-03)
+        bot = @horner(t , 1.0 , .273076135303957E+00 , .559398236957378E-01)
+        w = top/bot
+        return rangereduce ? t*w/a : a*(w+1)
+    else
+        top = @horner(t , .577215664901533E+00 , -.409078193005776E+00 , -.230975380857675E+00 , .597275330452234E-01 , .766968181649490E-02 , -.514889771323592E-02 , .589597428611429E-03)
+        bot = @horner(t , 1.0 , .427569613095214E+00 , .158451672430138E+00 , .261132021441447E-01 , .423244297896961E-02) 
+        w = top/bot
+        return rangereduce ? (t/a)*(w - 1.0) : a*w
+    end
+end
+"""
+    rgammax(a,x)
+
+Evaluation of exp(-x)*x^a/gamma(a) : ``1/\\Gamma(a) e^{-x} x^{a}``
+"""
+function rgammax(a::Float64,x::Float64)
+    if x == 0.0
+        return 0.0
+    elseif a >= 20.0
+        u =x/a
+        if u == 0.0
+            return 0.0
+        end
+        t = (1.0/a)^2
+        t1 = (((0.75*t - 1.0)*t + 3.5)*t - 105.0)/(a*1260.0) #Using Stirling Series : https://dlmf.nist.gov/5.11.1
+        t1 = t1 + a*logmxp1(u)
+        if t1 >= exparg
+            return rt2pin*sqrt(a)*exp(t1)
+        end
+    else
+        t = a*log(x) - x
+        if t < exparg
+            return 0.0
+        elseif a >= 1.0
+            return exp(t)/gamma(a)
+        else
+            return (a*exp(t))*(1.0 + rgamma1pm1(a))
+        end
+    end
+end   
+
+# Reference : 'Computation of the incomplete gamma function ratios and their inverse' by Armido R DiDonato , Alfred H Morris.
+# Published in Journal: ACM Transactions on Mathematical Software (TOMS)
+# Volume 12 Issue 4, Dec. 1986 Pages 377-393
+# doi>10.1145/22721.23109
+
+"""
+    gamma_p(a,x,IND)
+    
+    DLMF: https://dlmf.nist.gov/8.2#E4 , https://dlmf.nist.gov/8.2#E5
+    Wiki: https://en.wikipedia.org/wiki/Incomplete_gamma_function
+    IND --> Accuracy desired ; IND=0 means 14 significant digits accuracy , IND=1 means 6 significant digit and IND=2 means only 3 digit accuracy suffices.
+    gamma_p(a,x) or P(a,x) is the Incomplete gamma function ratio given by : ``1/\\Gamma (a) \\int_{0}^{x} e^{-t}t^{a-1} dt``
+"""
+function gamma_p(a::Float64,x::Float64,ind::Integer)
+    iop = ind + 1
```

Comment:
I would prefer to get rid of `iop`, and define methods for `Float32` instead.

Reply:
Keeping this for the last. Since, changes will be reflected for both. Hence, after this is finalised I'll just make a separate copy method for `Float32` and other.